File: SoccerPlayerApi/clientapp/src/app/models/aggregations/TimeAggregationDto.ts
import { TimeLevelDto } from "../levels/TimeLevelDto";

export class TimeAggregationDto {
    timeAggregationId: number;
    timeLevelId: number;
    timeLevel: TimeLevelDto;

    motherAggregationId?: number;
    motherAggregation?: TimeAggregationDto;

    label: string = '';

    constructor(
        timeAggregationId: number,
        timeLevelId: number,
        timeLevel: TimeLevelDto,
        motherAggregationId?: number,
        motherAggregation?: TimeAggregationDto,
        label: string = ''
    ) {
        this.timeAggregationId = timeAggregationId;
        this.timeLevelId = timeLevelId;
        this.timeLevel = timeLevel;
        this.motherAggregationId = motherAggregationId;
        this.motherAggregation = motherAggregation;
        this.label = label;
    }
}

File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Aggregations/TimeAggregationDto.cs
using SoccerPlayerApi.Dtos.Levels;

namespace SoccerPlayerApi.Dtos.Aggregations;

public class TimeAggregationDto
{
    public int TimeAggregationId { get; set; }

    public int TimeLevelId { get; set; }
    public TimeLevelDto TimeLevel { get; set; }

    public int? MotherAggregationId { get; set; }
    public TimeAggregationDto? MotherAggregation { get; set; }

    public string Label { get; set; } = string.Empty;
}
